### PR TITLE
Add support for reading and writing JSON

### PR DESCRIPTION
**Pull request by andrewpollock**
Reviewer feedback on https://github.com/google/osv.dev/pull/738 indicated this is the preferred format.

### CODE REVIEW

File: vulnfeeds/vulns/vulns.go [hunk 1]
```diff
@@ -364,3 +365,19 @@ func (v *Vulnerability) ToYAML(w io.Writer) error {
...
+	err := decoder.Decode(&vuln)
+	if err != nil {
+		return nil, err
+	}
+
+	return &vuln, nil
+}
+
+func (v *Vulnerability) ToJSON(w io.Writer) error {
+	encoder := json.NewEncoder(w)
```
Comment (another-rex) [inline vulnfeeds/vulns/vulns.go:381]: Can you add 
```
encoder.SetIndent("", "  ")
```
it makes the output much easier to read, which helps to check if output is correct.

**Review by oliverchang — APPROVED** [review]

File: vulnfeeds/vulns/vulns.go (hunk 1 above)
Comment (oliverchang) [reply vulnfeeds/vulns/vulns.go:381]: Can we at least make this optional? formatting can be easily done by e.g. using `'cat blah.json | jq`.

Comment (andrewpollock) [reply vulnfeeds/vulns/vulns.go:381]: Sure, done.

Comment (andrewpollock) [reply vulnfeeds/vulns/vulns.go:381]: Okay, conflicting requests. @another-rex are you happy to just post-process it for human readability purposes? If so, I'll back [e9a140e](https://github.com/google/osv.dev/pull/748/commits/e9a140ec9515d7e75894fcdd4e871ff31edd24cf) out.

Comment (another-rex) [reply vulnfeeds/vulns/vulns.go:381]: Sure, let's just post process it.

**Review by another-rex — APPROVED** [review]

**Pull request merged**